pkg/controllers/dynakube/extension/otel/statefulset.go

```diff
@@ -224,6 +222,12 @@ func setImagePullSecrets(imagePullSecrets []corev1.LocalObjectReference) func(o
 }
 
 func setVolumes(dk *dynakube.DynaKube) func(o *appsv1.StatefulSet) {
+	tlsSecretName := dk.Name + consts.ExtensionsTlsSecretSuffix
+
+	if tlsSecretName != "" {
```

Comment:
looks wrong, you just set it to constant on line 225, but now you are trying to compare with empty string 🤔  am I miss something?

Reply:
Yes, missed it :D

Fixed